pkg/apis/core/v1/defaults.go

```diff
@@ -181,6 +181,10 @@ func SetDefaults_PodSpec(obj *v1.PodSpec) {
 	if obj.SchedulerName == "" {
 		obj.SchedulerName = v1.DefaultSchedulerName
 	}
+	if obj.HostUser == nil && utilfeature.DefaultFeatureGate.Enabled(features.UserNamespaceRemapping) {
+		trueObj := true
+		obj.HostUser = &trueObj
```

Comment:
I didn't expect this to default to true... according to @derekwaynecarr there's no guarantee runtimes are giving containers host user namespace uids today

Reply:
From CRI API, this PR is sending "NODE" namespace option to runtime and therefore it is expected that runtimes will provide host usernamespace by default

Reply:
i think we need to discuss that more.  what happens if you run kata or gvisor in future?

Reply:
To handle runtimes like kata or gvisor, one more boolean field like `HostUserNamespaceSharingCapability` will be required in CRI API. This will be needed to differentiate between the following cases:
1. docker/crio type runtimes and usernamespace remapping is not enabled
2. Runtimes like gvisor and katacontainers

Complete behavior will be like this:

| Enabled at Kubelet| *hostUser   |  HostUserNamespace-SharingCapability | Enabled at runtime  |  UserNamespaceOption sent to runtime |
|---|---|---|---|---|
| false  |  non-nil |  * | *  | Validation Error  |
| *  | nil  | true  | true   | NODE_WIDE_REMAPPED  |
| *  | nil  |  true | false  | NODE  |
| true  | true  | false  | *  | validation error  |
| true  | false  | false  | *  | POD  |
|  true |  true | true  | *  | NODE  |
| true | false | true | false | validation error |